x-pack/plugins/ml/public/jobs/new_job/simple/components/enable_model_plot_checkbox/enable_model_plot_checkbox_directive.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { EnableModelPlotCheckbox } from './enable_model_plot_checkbox_view.js';
+import { ml } from 'plugins/ml/services/ml_api_service';
+
+import { uiModules } from 'ui/modules';
+const module = uiModules.get('apps/ml');
+
+module.directive('mlEnableModelPlotCheckbox', function () {
+  return {
+    restrict: 'AE',
+    replace: false,
+    scope: {
+      formConfig: '=',
+      ui: '=ui',
+      getJobFromConfig: '='
+    },
+    link: function ($scope, $element) {
+      const STATUS = {
+        FAILED: -1,
+        NOT_RUNNING: 0,
+        RUNNING: 1,
+        FINISHED: 2,
+        WARNING: 3,
+      };
+      const errorHandler = (error) => {
+        console.log('Cardinality could not be validated', error);
+        $scope.ui.cardinalityValidator.status = STATUS.FAILED;
+        $scope.ui.cardinalityValidator.message = 'Cardinality could not be validated';
+      };
+
+      // Only model plot cardinality relevant
+      // format:[{id:"cardinality_model_plot_high",modelPlotCardinality:11405}, {id:"cardinality_partition_field",fieldName:"clientip"}]
+      const checkCardinalitySuccess = (data) => {
+        const response = {
+          success: true,
+        };
+        // There were no fields to run cardinality on.
+        if (Array.isArray(data) && data.length === 0) {
+          return response;
+        }
+
+        for (let i = 0; i < data.length; i++) {
+          if (data[i].id === 'success_cardinality') {
+            break;
+          }
+
+          if (data[i].id === 'cardinality_model_plot_high') {
+            response.success = false;
+            response.highCardinality = data[i].modelPlotCardinality;
+            break;
+          }
+        }
+
+        return response;
+      };
+
+      const validateCardinality = function () {
+        $scope.ui.cardinalityValidator.status = STATUS.RUNNING;
```

Comment:
this isn't triggering the `$watch` when toggling the checkbox.
This is due to the toggle happing in react, so the call to this function isn't trigging the digest cycle.
I'd suggest adding a `$scope.$applyAsync();` call to the bottom of this function and to the bottom of the `(response) => {` function

Reply:
Hey I'm not quite sure what you mean it isn't triggering the `$watch` - when I tested it I can see the copy of the checkbox changing to `Validating cardinality...` as it should when `$scope.ui.cardinalityValidator.status` is updated to `RUNNING`. I _did_ have to put a setTimeout on it to test it though since it happens so fast. Is that what you're referring to?

Reply:
I'll try to reproduce on a local checkout. IMHO if we need to use `$applyAsync` then all the code needs to be wrapped in the callback of `$applyAsync`, not the other way around :)